### DISCUSSION TITLE
Use more CPU threads

### DISCUSSION THREAD

**Discussion by ghost** [AbdBarho/stable-diffusion-webui-docker#419] [General]
I am running this on my server which has a Ryzen 5600U (no GPU ATM). When generating an image, only 6 threads are used despite 10 being allocated. What is the easiest way to change this? I've tried setting environment variables `OMP_NUM_THREADS` and `MKL_NUM_THREADS` to no avail. It appears I could also use [this function](https://pytorch.org/docs/stable/generated/torch.set_num_threads.html), but I'm not sure where to put it (I have very little experience with Python).

**Comment by AbdBarho** [comment]
I think this is something that should be asked in the original repo of whichever UI you are using.

**Comment by ghost** [comment]
I'm using the auto-cpu UI. How can I modify the UI's files in a way that will persist across rebuilds?

**Reply by AbdBarho** [reply]
what UI files exactly?

**Reply by ghost** [reply]
`/modules/devices.py`

**Reply by AbdBarho** [reply]
hmmmm, I have never used this before, can you explain what it is, and how can I use it? it would make it a lot easier to add support for it.

If there are any links to documentation where that is described that would be great.

**Reply by ghost** [reply]
It is not an official feature, but rather a patch suggested here:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3387#issuecomment-1296340015

**Reply by AbdBarho** [reply]
Aha, I see.

Here is a hack with bash after quick google search, I have not tried it, but you want should be similar
```sh
LINE='torch.set_num_threads(8);torch.set_num_interop_threads(8);'
FILE='/stable-diffusion/modules/devices.py'
grep -qF -- "$LINE" "$FILE" || echo "$LINE" >> "$FILE"
```

you can add this to your startup script.

**Comment by ghost** [comment]
I got it working with this script:
```sh
THREADS="10"
LINE="torch.set_num_threads(${THREADS});torch.set_num_interop_threads(${THREADS});"
PATTERN="set_num_threads"
FILE="/stable-diffusion-webui/modules/devices.py"
grep -qF "$PATTERN" "$FILE" || sed -i "79 a $LINE" $FILE
```
